docs/network-policy.md

```diff
@@ -0,0 +1,70 @@
+# NetworkPolicy
+Kubernetes NetworkPolicy documentation: https://kubernetes.io/docs/concepts/services-networking/network-policies/#behavior-of-to-and-from-selectors
+
+Each NetworkPolicy object consists of three sections:
+
+1. `podSelector`: a label selector that determines which pods the NetworkPolicy applies to
+2. `ingress`: determines which entities ingress traffic can be received from by the pods selected by the NetworkPolicy
+3. `egress`: determines which entities egress traffic can be sent to by pods selected by the NetworkPolicy
+
+## Unicast default-deny
+When a pod is selected by one or more NetworkPolicies it becomes isolated and all unicast ingress and egress traffic is blocked. Two global PortGroups are used to facilitate this behavior: `ingressDefaultDeny` and `egressDefaultDeny`.  Any pod selected by a NetworkPolicy in any namespace is added to these PortGroups.
+
+Two ACLs (four total) are added to each PortGroup:
+
+1. a drop policy with `priority=1000` and `direction=to-lport`
+2. an allow policy for ARP traffic with `priority=1001`, `direction=to-lport`, and `match=arp`
+
+## Multicast
+All multicast traffic is blocked by default, and must be enabled on a per-Namespace basis using an annotation on that Namespace.
+
+### default-deny
+Multicast traffic is blocked by the `mcastPortGroupDeny` PortGroup, which all pods are added to. Two ACLs are added to the PortGroup:
+
+1. a drop policy with `priority=1011`, `direction=from-lport`, and `match=ip4.mcast`
+2. a drop policy with `priority=1011`, `direction=to-lport`, and `match=ip4.mcast`
+
+#### allow
+To allow multicast traffic for a given Namespace, a PortGroup is created for that namespace with the name `mcastPortGroup-FOO` where FOO is the Namespace's name. Each pod created in a Namespace is added to this PortGroup.
+
+Two ACLs are added to the Namespace's multicast allow port group;
+
+1. an allow policy with `priority=1012`, `direction=from-lport`, and `match=ip4.mcast`
+2. an allow policy with `priority=1012`, `direction=to-lport`, and `match=ip4.mcast`
+
+## Namespace AddressSet
+Each Namespace creates an AddressSet to which the IPs of all pods in that Namespace are added. This is used in NetworkPolicy `ingress` and `egress` sections to implement the namespace selector.
+
+## `podSelector`
+The `podSelector` is a label selector, which can be either a list of labels (`app=nginx`) or a match expression. Only pods in the same namespace as the NetworkPolicy can be selected by it. The end result is a list of zero or more pods to which this NetworkPolicy's `ingress` and `egress` sections will be applied.
+
+By default pods may send to an recipient and receive from any source. When a pod is selected by one or more NetworkPolicies it becomes isolated and all traffic is blocked. NetworkPolicy may then be used to allow received traffic from sources determined by the `ingress` section and/or send traffic to recipients specified by the `egress` section.
```

Comment:
typo "anY recipient"

again, isolation description fails to distinguish "isolated for ingress" and "isolated for egress"